src/api/verify.ts

```diff
@@ -78,14 +83,86 @@ class VerifyAPI {
       expiresAt,
       expiresIn,
       passed,
+      failureReasons,
+      _id,
     } as RadarTrackVerifiedResponse;
 
     if (options.debug) {
       trackRes.response = response;
     }
 
+    lastToken = trackRes;
+    lastTokenNow = performance.now();
+
+    if (tokenCallback) {
+      tokenCallback(trackRes);
+    }
+
     return trackRes;
   }
+
+  static async startTrackingVerified(params: RadarStartTrackingVerifiedParams) {
+    const doTrackVerified = async () => {
+      const trackRes = await this.trackVerified({});
+
+      const { interval } = params;
+
+      let expiresIn = 0;
+      let minInterval = interval;
+
+      if (trackRes) {
+        expiresIn = (trackRes.expiresIn || expiresIn);
+
+        // if expiresIn is shorter than interval, override interval
+        minInterval = Math.min(expiresIn, interval);
+      }
+
+      // re-request early to maximize the likelihood that a cached token is available
+      if (minInterval > 20) {
+        minInterval = minInterval - 10;
+      }
+
+      // min interval is 10 seconds
+      if (minInterval < 10) {
+        minInterval = 10;
```

Comment:
so if I give the interval param as 21 then it is effectively 11 but if I give 15 it is effectively 15?

Reply:
Yeah. I think that's an edge case, but if it bothers us, I can update